pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -902,10 +920,6 @@ func (r *ReconcileClusterPool) assignClustersToClaims(claims []*hivev1.ClusterCl
 			claim.Spec.Namespace = cds[0].Namespace
 			cds = cds[1:]
 			logger.WithField("cluster", claim.Spec.Namespace).Info("assigning cluster to claim")
-			if err := r.Update(context.Background(), claim); err != nil {
-				logger.WithError(err).Log(controllerutils.LogLevel(err), "could not assign cluster to claim")
-				return cds, err
-			}
```

Comment:
hmm.. it seems like removing this would stop assigning clusters to claim, as this updates the spec ??

Reply:
I removed this update because we are calling another update right below `else` - see https://github.com/openshift/hive/blob/9e0a375372593f2bb7890f7cfee4a615ed2daff9/pkg/controller/clusterpool/clusterpool_controller.go#L943-L949

Reply:
the updates calls are not the same and that's why i was concerned.

the removed code updates the spec.
```
if err := r.Update(context.Background(), claim); err != nil {
```

while the linked code in previous comment only updates the status.
```
if err := r.Status().Update(context.Background(), claim); err != nil { 
```

it's weird that no unit test is failing because of this..

Reply:
Ah - I didn't notice - yes - it is weird, but I will make the changes to not drop the update code here...

Reply:
done - thank you for catching that.
Looks like for the test that checks if claim is assigned is passing regardless of whether I keep or remove that update :shrug: 
Here's the [check](https://github.com/openshift/hive/blob/3c57723f538887a124852bffd9b7dd0d9070c852/pkg/controller/clusterpool/clusterpool_controller_test.go#L986-L995) and here's the [test](https://github.com/openshift/hive/blob/3c57723f538887a124852bffd9b7dd0d9070c852/pkg/controller/clusterpool/clusterpool_controller_test.go#L583L597)